internal/server/manager.go

```diff
@@ -37,18 +38,32 @@ type ClusterInfo struct {
 	FailureMessage *string           `json:"failureMessage"`
 }
 
-type instance struct {
-	client     client.Client
-	scheme     *runtime.Scheme
-	clusterMux sync.Mutex // use a Mutex to update managed Clusters
+type ClusterProfileStatus struct {
+	Name        *string                         `json:"name"`
+	Namespace   *string                         `json:"namespace"`
+	ClusterName *string                         `json:"clusterName"`
+	Summary     []configv1alpha1.FeatureSummary `json:"summary"`
```

Comment:
nit:FeatureSummary has few extra fields we don't need here. Essentially here we need the FeatureID, Status and the FailureMessage.

That will help the amount of info we keep in memory.